day3/day3.py

```diff
@@ -0,0 +1,75 @@
+import logging
+import re
+from typing import List, Union, Tuple
+
+import numpy as np
+
+import day2
+
+
+def parse_claims(mylines: Union[List[str], Tuple[str]]):
+    '''
+    Parses the elves claims
+
+    :param mylines: A list object containing the fabric claims as separate entities
+    to be parsed in the format (#int @ int,int: int x int)
+    :return: For each entity in the original list this function converts them to [['str','str','str','str']]
+    '''
+    if not mylines:
+        raise ValueError("mylines can not be empty")
+
+    if type(mylines) not in {tuple, list}:
+        raise TypeError("mylines must be tuple, list, or generator")
+
+    fabric_coordinates = []
+    for i in mylines:
+        # gets rid of the empty spaces
+        j = i.replace(' ', '')
+        parsed_claim = re.split('#|@|,|:|x', j)
+        # remove the #claim
+        fabric_coordinates.append(parsed_claim[1:6])
+    return fabric_coordinates
+
+
+def parse_fabric_list(claims, size):
+    '''
+    Puts the elves claims into an array object
+
+    :param claims: a list object made up of lists in the format [['str','str','str','str'],['str','str','str','str']]
+    :return: the overall number of inches (int) which overlap in the making of Santa's magic suit
+    '''
+    santa_fabric = np.zeros((size, size))
+    for i in claims:
+        # converting the list of strings to int
+        i = list(map(int, i[1:5]))
+        santa_fabric[i[1]:i[1] + i[3], i[0]:i[0] + i[2]] += 1
+    return np.sum(santa_fabric > 1), santa_fabric
+
+
+def uncontested_claim(claims, fabric_map: list) -> int:
+    '''
+    Checks for claims in the fabric which do not overlap with other claims
+
+    :param fabric_map: an array containing the submitted elf claims for santa's fabric:
```

Comment:
add docstring for parameter `claim`
remember:
1. update code
2. update docstring
3. update test